app/services/katello/pulp3/repository.rb

```diff
@@ -111,6 +115,17 @@ def get_distribution(href)
         nil
       end
 
+      def update_distributions
+        paths.values.select { |prefix, _path| prefix == 'http' }.each do |path|
+          dists = lookup_distributions(base_path: path.sub(/^\//, ''))
+          delete_distribution(dists.first._href) if dists.first
+
+          if repo.root.unprotected
+            create_distribution(prefix, path)
+          end
+        end
+      end
```

Comment:
Since you asked for comments around testing, this might be the only thing i see here currently that i might would unit test specifically with stubbed api calls